packages/api/src/controllers/access-control.ts

```diff
@@ -242,6 +243,24 @@ app.post(
       return res.end();
     }
 
+    let origin: string = null;
+    if (req.body?.webhookPayload?.headers) {
+      origin = req.body?.webhookPayload?.headers["origin"];
+    }
```

Comment:
nit: could do in one line if you want
```suggestion
    const origin = req.body?.webhookPayload?.headers?.origin;
```